src/app/clusters/power-source-configuration-server/power-source-configuration-server.cpp

```diff
@@ -61,9 +64,27 @@ CHIP_ERROR PowerSourceConfigurationAttrAccess::Read(const ConcreteReadAttributeP
     {
     case Sources::Id:
         err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+            std::pair<uint16_t, uint8_t> orderEpPair[kMaxPowerSources];
+            uint8_t idx = 0, order = 0;
             for (auto endpoint : EnabledEndpointsWithServerCluster(PowerSource::Id))
             {
-                ReturnErrorOnFailure(encoder.Encode(endpoint));
+                if (idx >= kMaxPowerSources)
+                    break;
+                PowerSource::Attributes::Order::Get(endpoint, &order);
+                orderEpPair[idx] = std::make_pair(endpoint, order);
+                idx++;
+            }
+
+            int i, j, k;
```

Comment:
Please just declare the loop vars in the relevant loop headers, so they don't leak out of the loops?